recipes/libvpx/all/conanfile.py

```diff
@@ -0,0 +1,151 @@
+    topics = ("conan", "vpx", "codec", "web", "VP8", "VP9")
+    license = "BSD-3-Clause"
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'fPIC': True}
+
+    _source_subfolder = "source_subfolder"
+
+
+    def configure(self):
+        if self.settings.os == 'Windows' and self.options.shared:
+            raise ConanInvalidConfiguration('Windows shared builds are not supported')
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires('yasm/1.3.0')
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires('msys2/20200517')
+
+    def source(self):                                                                                    
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _fix_sources(self):
+        gen_vcxproj = os.path.join(self._source_subfolder, 'build', 'make', 'gen_msvs_vcxproj.sh')
+        tools.replace_in_file(gen_vcxproj,
+                              '        --help|-h) show_help',
+                              '        --help|-h) show_help\n        ;;\n'
+                              '        -O*) echo "ignoring $opt..."\n')
+        tools.replace_in_file(gen_vcxproj,
+                              '        --help|-h) show_help',
+                              '        --help|-h) show_help\n        ;;\n'
+                              '        -Zi) echo "ignoring $opt..."\n')
+        tools.replace_in_file(gen_vcxproj,
+                              '        --help|-h) show_help',
+                              '        --help|-h) show_help\n        ;;\n'
+                              '        -MD*|-MT*) echo "ignoring $opt (this option is already handled)"\n')
+        # disable warning:
+        # vpx.lib(vpx_src_vpx_image.obj) : MSIL .netmodule or module compiled with /GL found; restarting link
+        # with /LTCG; add /LTCG to the link command line to improve linker performance
+        tools.replace_in_file(gen_vcxproj,
+                              'tag_content WholeProgramOptimization true',
+                              'tag_content WholeProgramOptimization false')
+        msvs_common_sh = os.path.join(self._source_subfolder, "build", "make", "msvs_common.sh")
+        tools.replace_in_file(msvs_common_sh,
+                              'if [ "$(uname -o 2>/dev/null)" = "Cygwin" ] \\',
+                              'if [ "$(uname -o 2>/dev/null)" = "Cygwin" ] || [ "$(uname -o 2>/dev/null)" = "Msys" ]\\')
+
+    def _configure_autotools(self):
```

Comment:
Hillarious build system... Thought that autotools is hard :)

Reply:
it would be much easier if they just used auto-tools...